lib/ansible/module_utils/basic.py

```diff
@@ -1646,14 +1646,14 @@ def atomic_move(self, src, dest, unsafe_writes=False, keep_dest_attrs=True):
                             os.close(tmp_dest_fd)
                             # leaves tmp file behind when sudo and not root
                             try:
-                                shutil.move(b_src, b_tmp_dest_name, copy_function=shutil.copy2 if keep_dest_attrs else shutil.copy)
+                                shutil.move(b_src, b_tmp_dest_name, copy_function=shutil.copy if keep_dest_attrs else shutil.copy2)
```

Comment:
I've been wondering if we should add another kwarg to atomic_move, to let third parties do this. I can see why it's counterintuitive if you're not reading the method understanding that this is handling an internal use case.

Reply:
I'll leave that to a subsequent PR if we decide this should be something configurable above the current 'keep' option